### PR TITLE
Improve 404 messages for misspelled repo names/owners or lack of permissions

### LINKED ISSUE #271: Improve error messages when repository not found
Currently it looks like this

``` r
remotes::install_github("r-lib/arst")
#> Error: HTTP error 404.
#>   Not Found
#> 
#>   Rate limit remaining: 4990/5000
#>   Rate limit reset at: 2019-01-04 13:57:37 UTC
#> 
#> 
```

<sup>Created on 2019-01-04 by the [reprex package](https://reprex.tidyverse.org) (v0.2.1)</sup>

Some ideas for improvement

- Only show the rate limit message for 403 errors or when the rate limit is 0 (which should be a 403 error)
- Add a hint when it is a 404 error that it is likely due to a misspelled repo

The relevant code is https://github.com/r-lib/remotes/blob/a0d6cd5d52062fcaedac7dd19b03f9211f578791/R/github.R#L144-L188

### LINKED ISSUE #276: install_github 404 error when Suggest'ed remote is private
I maintain a package to which I've added a `Suggest`ed package that is in a private repository. Even though the default is to not install `Suggest`ed packages, I still receive a 404 error with the following.

```
devtools::install_github("Displayr/flipFormat")
```

I would expect this to work without error, but it would also be nice if the error message could be improved to state the offending package (#271).

### PR DESCRIPTION
**Pull request by tylermorganwall**
I have included code to return the repo name and repo owner in the error message when the status code is 404 (to check for misspellings), as well as remind the user to check if they have the required permissions to access the repo.

Fixes #271.
Fixes #276.

### CODE REVIEW

**Comment by gaborcsardi** [comment]
Thanks! Taking a look.....

File: R/github.R [hunk 1]
```diff
@@ -165,8 +165,28 @@ github_error <- function(res) {
           "Use `usethis::edit_r_environ()` and add the token as `GITHUB_PAT`."
         }
       )
+  } else if (identical(as.integer(res$status_code), 404L)) {
+    repo_information = re_match(res$url,"(repos)/(?P<owner>[^/]+)/(?P<repo>[^/]++)/")
```
Comment (gaborcsardi) [inline R/github.R]: Can you pls use `<-` and space after `,`?

File: R/github.R [hunk 2]
```diff
@@ -165,8 +165,28 @@ github_error <- function(res) {
...
+    pat_guidance <- sprintf(
```
Comment (gaborcsardi) [inline R/github.R]: Can you rename `pat_guidence` to just `guidence`. It is now not always about the PAT.

File: R/github.R (hunk 1 above)
Comment (tylermorganwall) [reply R/github.R]: Done.

File: R/github.R (hunk 2 above)
Comment (tylermorganwall) [reply R/github.R]: Done.

**Comment by gaborcsardi** [comment]
Looks good, thanks much!

**Comment by QuLogic** [comment]
A few files were inadvertently made executable here.

**Comment by gaborcsardi** [comment]
@QuLogic can you pls open an issue or PR? Thanks!

**Comment by QuLogic** [comment]
OK #341

**Pull request merged**